src/include/duckdb/execution/index/art/const_prefix_handle.hpp

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/execution/index/art/const_prefix_handle.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "duckdb/execution/index/fixed_size_allocator.hpp"
+#include "duckdb/execution/index/art/art.hpp"
+#include "duckdb/execution/index/art/node.hpp"
+
+namespace duckdb {
+
+//! ConstPrefixHandle is a read-only wrapper to access a prefix node.
+//! A segment handle is used for memory management, but it is not marked as modified.
+class ConstPrefixHandle {
+public:
+	static constexpr NType PREFIX = NType::PREFIX;
+
+	ConstPrefixHandle() = delete;
+	ConstPrefixHandle(const ART &art, const Node node);
+	ConstPrefixHandle(const ConstPrefixHandle &) = delete;
+	ConstPrefixHandle &operator=(const ConstPrefixHandle &) = delete;
+	ConstPrefixHandle(ConstPrefixHandle &&) = delete;
+	ConstPrefixHandle &operator=(ConstPrefixHandle &&) = delete;
+
+public:
+	uint8_t GetCount(const ART &art) const;
+	uint8_t GetByte(const idx_t pos) const;
+
+	static inline uint8_t Count(const ART &art) {
+		return art.prefix_count;
+	}
```

Comment:
I wonder if this should be a `const` function on the `ART` with `prefix_count` becoming private... something like `art.GetPrefixCount()`?